### DISCUSSION TITLE
Feature suggestion : Wps password grabber.

### DISCUSSION THREAD

**Discussion by Z-SHOCK-E** [BruceDevices/firmware#1281] [General]
It will be nice to grab wifi password using wps. 

https://www.esp32.com/viewtopic.php?t=27004
https://github.com/espressif/arduino-esp32/blob/master/libraries/WiFi/examples/WPS/WPS.ino

**Comment by Z-SHOCK-E** [comment]
Esp32 wifi wps(push button) can be used to authenticate with a router and save the password as a test file.

**Comment by zcaliptium** [comment]
Today I dig into provided examples and tried to adapt it for Bruce... And found some caveat.

The Bruce firmware uses really outdated ESP-IDF version (4.4.7). So it lacks of full WPS functionality.

There is ``esp_wps_config_t`` structure that passed into ``esp_wifi_wps_enable`` function.

The main problem In ``4.4.7`` it just haven't field for the PIN code:
```cpp
typedef struct {
    wps_type_t wps_type;
    wps_factory_information_t factory_info;
} esp_wps_config_t;
```

If we check docs for the 5.0+ version we will find that it suddenly appears:
```cpp
typedef struct {
    wps_type_t wps_type;
    wps_factory_information_t factory_info;
    char pin[PIN_LEN]
} esp_wps_config_t;
``` 

https://docs.espressif.com/projects/esp-idf/en/release-v5.0/esp32/api-reference/network/esp_wifi.html#_CPPv416esp_wps_config_t
